fairlearn/metrics/_annotated_metric_function.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (c) Fairlearn contributors.
+# Licensed under the MIT License.
+
+from typing import Callable, Dict, List, Optional
+
+import logging
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+_DEFAULT_NAME = "metric"
+
+_METRIC_FUNCTION_NONE = "Found 'None' instead of metric function"
+_METRIC_FUNCTION_NOT_CALLABLE = "Object passed as metric function not callable"
+
+
+class AnnotatedMetricFunction:
+    """Wrapper for functions, to give them args and kwargs."""
```

Comment:
This doesn't seem quite the right description. My sense is that the goal of this wrapper is so that it can be applied to a `DataFrame`. So, if anything, the class should be probably called `DataFrameMetric` or something like that?